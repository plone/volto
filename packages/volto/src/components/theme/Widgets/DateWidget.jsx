import React from 'react';
import cx from 'classnames';
import { injectLazyLibs } from '@plone/volto/helpers/Loadable/Loadable';
import { useSelector } from 'react-redux';
import { toBackendLang } from '@plone/volto/helpers';

const DateWidget = ({ value, children, className, format = 'll', moment }) => {
  const lang = useSelector((state) => state.intl.locale);
  const _moment = moment.default;
  _moment.locale(toBackendLang(lang));
  return value ? (
    <span className={cx(className, 'date', 'widget')}>
      {children
        ? children(_moment(value).format(format))
        : _moment(value).format(format)}
    </span>
  ) : (
    ''
  );
};

export default injectLazyLibs(['moment'])(DateWidget);
