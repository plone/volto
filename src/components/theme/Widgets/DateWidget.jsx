import React from 'react';
import cx from 'classnames';
import moment from 'moment';
import { useSelector } from 'react-redux';
import { toBackendLang } from '@plone/volto/helpers';

const DateWidget = ({ value, children, className, format = 'll' }) => {
  const lang = useSelector((state) => state.intl.locale);
  moment.locale(toBackendLang(lang));
  return value ? (
    <span className={cx(className, 'date', 'widget')}>
      {children
        ? children(moment(value).format(format))
        : moment(value).format(format)}
    </span>
  ) : (
    ''
  );
};

export default DateWidget;
