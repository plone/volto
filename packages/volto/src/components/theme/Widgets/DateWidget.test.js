import React from 'react';
import renderer from 'react-test-renderer';
import DateWidget from './DateWidget';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);
jest.mock('@plone/volto/helpers/Loadable/Loadable');
beforeAll(
  async () =>
    await require('@plone/volto/helpers/Loadable/Loadable').__setLoadables(),
);

const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
});

describe('DateWidget', () => {
  it('renders an empty date view widget component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <DateWidget />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a date view widget component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <DateWidget className="metadata" value="2020-08-18" />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a date view widget component with L format', () => {
    const component = renderer.create(
      <Provider store={store}>
        <DateWidget className="metadata" value="2020-08-18" format="L" />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a date view widget component with children', () => {
    const component = renderer.create(
      <Provider store={store}>
        <DateWidget className="metadata" value="2020-08-18">
          {(child) => <strong>{child}</strong>}
        </DateWidget>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
