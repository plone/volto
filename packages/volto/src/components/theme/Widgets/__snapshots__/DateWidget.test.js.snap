// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateWidget renders a date view widget component 1`] = `
<span
  className="metadata date widget"
>
  Aug 18, 2020
</span>
`;

exports[`DateWidget renders a date view widget component with L format 1`] = `
<span
  className="metadata date widget"
>
  08/18/2020
</span>
`;

exports[`DateWidget renders a date view widget component with children 1`] = `
<span
  className="metadata date widget"
>
  <strong>
    Aug 18, 2020
  </strong>
</span>
`;

exports[`DateWidget renders an empty date view widget component 1`] = `""`;
